api/v1alpha1/envoygateway_types.go

```diff
@@ -271,7 +271,11 @@ type EnvoyGatewayCustomProvider struct {
 	// This provider is used to specify the provider to be used
 	// to provide an environment to deploy the out resources like
 	// the Envoy Proxy data plane.
-	Infrastructure EnvoyGatewayInfrastructureProvider `json:"infrastructure"`
+	//
+	// Infrastructure is optional, if no provider is set,
+	// The Kubernetes will be used as the default infrastructure provider.
```

Comment:
we should make this explicit, and not default to k8s here, because users have expressed a desire for EG to not provision any infra, more in https://envoyproxy.slack.com/archives/C03E6NHLESV/p1722535811941329?thread_ts=1722481296.411989&cid=C03E6NHLESV

Reply:
updated